### PR TITLE
feat: custom gas in all networks

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes BX-1546, BX-1507
Figma link (if any):

## What changed (plus any additional context for devs)

enabling custom gas in all networks for all txs types

## Screen recordings / screenshots

https://www.loom.com/share/19d39fed3c1040c29ae00a41eddc8a81

### CODE REVIEW

File: src/entries/popup/components/TransactionFee/CustomGasSheet.tsx [hunk 1]
```diff
@@ -196,12 +201,19 @@ export const CustomGasSheet = ({
...
+
   const [maxBaseFeeWarning, setMaxBaseFeeWarning] = useState<
     'stuck' | 'fail' | undefined
   >(undefined);
   const [maxPriorityFeeWarning, setPriorityBaseFeeWarning] = useState<
     'stuck' | 'fail' | undefined
   >(undefined);
+  const [gasPriceWarning, setGasPriceWarning] = useState<
+    'stuck' | 'fail' | undefined
+  >(undefined);
```
Comment (magiziz) [inline src/entries/popup/components/TransactionFee/CustomGasSheet.tsx:216]: Maybe we can do something like this ?

```tsx
const [gasPriceWarning, setGasPriceWarning] = useState<'stuck' | 'fail'>();
```

File: src/entries/popup/hooks/useGas.ts [hunk 2]
```diff
@@ -204,6 +218,43 @@ const useGas = ({
...
+  useEffect(() => {
+    if (
+      !gasData ||
+      !enabled ||
+      prevDebouncedGasPrice === debouncedGasPrice ||
+      feeType !== 'legacy' ||
+      !nativeAsset
+    ) {
+      return;
+    }
```
Comment (magiziz) [inline src/entries/popup/hooks/useGas.ts:230]: Can we use `useQuery` here ?

**Comment by estebanmino** [comment]
@magiziz I just applied your comments! 🙏

**Review by derHowie — APPROVED** [review]

**Comment by BrodyHughes** [comment]
Functionality seems good to me. FOr this sort of case we should do a '<0.0001' or something though:

Screenshot 2024-08-01 at 4 06 23 PM

**Comment by DanielSinclair** [comment]
@BrodyHughes Which network was that? Wonder if it is a bad result from Backend

**Review by DanielSinclair — APPROVED** [review]

**Comment by DanielSinclair** [comment]
Lgtm. Have a few design tweaks for existing Custom Gas UI problems, so will create a seperate polish ticket for that and carry over Brody's issue

**Pull request merged**